[nnc] Enable CPU fusion inside Facebook

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55622 [nnc] Enable CPU fusion inside Facebook**

We've been testing this for months, it's time.

Differential Revision: [D27652484](https://our.internmc.facebook.com/intern/diff/D27652484/)

bertmaher committed:
[nnc] Enable CPU fusion inside Facebook

We've been testing this for months, it's time.

Differential Revision: [D27652484](https://our.internmc.facebook.com/intern/diff/D27652484/)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good, but you ought to fix that windows build failure 😜

bertmaher committed:
rebase on "[nnc] Enable CPU fusion inside Facebook"


We've been testing this for months, it's time.

Differential Revision: [D27652484](https://our.internmc.facebook.com/intern/diff/D27652484/)

[ghstack-poisoned]

bertmaher committed:
Update on "[nnc] Enable CPU fusion inside Facebook"


We've been testing this for months, it's time.

Differential Revision: [D27652484](https://our.internmc.facebook.com/intern/diff/D27652484/)

[ghstack-poisoned]
